### DISCUSSION TITLE
Problem with [TestCaseSource] containing classes that override ToString()

### DISCUSSION THREAD

**Discussion by ClaudiuHBann** [nunit/nunit#4787] [Q&A] [answered]
Is there any known problem with the attribute `TestCaseSource` where the source contains an instance of a class that overrides the method `ToString` ?

**Comment by OsirisTerje** [comment]
None I am aware of.   What do you see?

**Reply by ClaudiuHBann** [reply]
I will come back with a full example about what I have encountered.

**Comment by manfred-brands** [comment, marked as answer]
To create the name of the test case, NUnit will for non-primitive instances call `ToString` on the object.
This could lead to unusual names if the output contains newlines.
However I tried to put in XML reserved characters (&) and it still worked fine.

**Reply by ClaudiuHBann** [reply]
My models had the method `ToString` implemented using the `JSON` pretty format.
Is there any workaround besides making a method with another name? Or it is what it is?

**Reply by manfred-brands** [reply]
You can wrap your `TestCaseSource` into `TestData` objects which has a `SetName` property:

```csharp
private static IEnumerable<TestCaseData> TestCaseArguments()
{
    yield return new TestCaseData(new MyTestObject(...)).SetName("Name A");
    yield return new TestCaseData(new MyTestObjecy(...)).SetName("Name B");
}
```

**Reply by ClaudiuHBann** [reply]
Interesting, thank you for helping me out guys, I appreciate it.
